pkg/chunk/schema.go

```diff
@@ -42,21 +42,30 @@ type hasChunksForIntervalFunc func(userID, seriesID string, from, through model.
 
 // Schema interface defines methods to calculate the hash and range keys needed
 // to write or read chunks from the external index.
-type Schema interface {
+type BaseSchema interface {
+	// When doing a read, use these methods to return the list of entries you should query
+	GetReadQueriesForMetric(from, through model.Time, userID string, metricName string) ([]IndexQuery, error)
+	GetReadQueriesForMetricLabel(from, through model.Time, userID string, metricName string, labelName string) ([]IndexQuery, error)
+	GetReadQueriesForMetricLabelValue(from, through model.Time, userID string, metricName string, labelName string, labelValue string) ([]IndexQuery, error)
+	FilterReadQueries(queries []IndexQuery, shard *astmapper.ShardAnnotation) []IndexQuery
+}
+
+// Schema used by store
```

Comment:
Should be something like the following? If so, please remember to update `SeriesStoreSchema` doc too:
`StoreSchema  is the schema used by store.`